### ISSUE TITLE
2.94: Very bad handling of the "No space left on device" situation

### ISSUE THREAD

**Issue by rico666** [transmission/transmission#1835]
As soon as this happens, transmission starts to try to download other torrents that were queued for download and somehow clogs up the space even more.

Then, the error "Unable to save resume file: no space left on device" hits every single torrent - even those who are completely downloaded.

It cannot even delete any torrents anymore - say you pick one and say "Trash Data and Remove Torrent" - it is able to delete it from the list (UI frontend), but the data still remain on HDD.

In other words: Once your disk is full, transmission 2.94 makes everything go to sh*t. Hopefully 3.xx is not as bad, but in case it is, you might want to look into "disk full" behavior.

**Comment by Petemir** [comment]
It would be ideal if instead of starting a torrent that would otherwise fill the remaining available storage, transmission started it paused.

**Comment by philliplongman** [comment]
Still seeing problems with this on 3.00 (f4489c982e). Torrents that were active when you run out of space never behave properly in the queue afterwards. Torrents that were partially complete have to be forced to resume. Torrents that have metadata but no data downloaded show the error that says their data has been moved (even though they didn't have any yet). They have to be removed and readded.

All of this could be avoided by simply pausing all the torrents when the disk reaches a pre-set amount of free space.

**Comment by killemov** [comment]
You can remove the 2.94 version number from the title as this has been the normal behavior since forever. THE core problem is that there is no free space detection at all. It is all event based handling starting with not being able to write new data to disk. There is some point to be made for you being responsible for having enough free space. But with sparse files it is very hard to do any calculations.

My advice would also be to introduce some low water mark (16MB but configurable) that stops all downloading torrents to a specific download location when that mark is reached. This virtually makes your drive 16MB smaller, but saves you the hassle of rechecking or even removing+re-adding the affected torrents. Also simply loading the torrent metadata from a magnet and then stop would also help.
